### PR TITLE
`Development`: Remove deprecated file upload endpoint

### PR DESCRIPTION
**Pull request by julian-christl**
#### General

- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [X] I added multiple integration tests (Spring) related to the features (with a high test coverage).
#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context

After replacing all uploads with multipart uploads, these two endpoints are not used anymore.

### Description

I removed the two endpoints from the server and all related tests. I also removed the endpoint from the client and simplified the corresponding service. I added some checks for the file retrieval to keep the test coverage stable.

### Review Progress

Confirm that the endpoints are not in use anymore.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java [hunk 1]
```diff
@@ -290,7 +248,11 @@ public ResponseEntity<byte[]> getFileUploadSubmission(@PathVariable Long exercis
             throw new AccessForbiddenException();
         }
 
-        return buildFileResponse(filePathService.actualPathForPublicPath(URI.create(submission.getFilePath())), false);
+        if (submission.getFilePath() == null) {
+            throw new EntityNotFoundException("Submission " + submissionId + " has no file");
+        }
+
+        return buildFileResponse(Objects.requireNonNull(FilePathService.actualPathForPublicPath(URI.create(submission.getFilePath()))), false);
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java:255]: You repeat this check and path construction quite a few times. Is it possible to move (some of) this into buildFileResponse or another helper method?

**Review by JohannesStoehr — APPROVED** [review]
Removal looks good to me. Left one optional comment

**Review by RY997 — APPROVED** [review]
LGTM

**Review by JohannesStoehr — DISMISSED** [review]
Code looks even better now

**Comment by julian-christl** [comment]
Thanks, I forgot to remove the redundant null-check, though.

**Review by JohannesStoehr — APPROVED** [review]

**Review by Strohgelaender — APPROVED** [review]
code

**Review by RY997 — APPROVED** [review]
Reapprove!

**Pull request merged**